packages/fast-usdc-contract/src/fast-usdc.flows.ts

```diff
@@ -11,3 +37,97 @@ export const makeNobleAccount = (async (orch: Orchestrator) => {
   return nobleChain.makeAccount();
 }) satisfies OrchestrationFlow;
 harden(makeNobleAccount);
+
+export const forwardFunds = async (
+  orch: Orchestrator,
+  {
+    currentChainReference,
+    supportsCctp,
+    log = makeTracer('FlowForwardFunds'),
+    getNobleICA,
+    settlementAccount,
+    statusManager,
+  }: Context,
+  tx: {
+    txHash: EvmHash;
+    amount: NatAmount;
+    destination: AccountId;
+  },
+) => {
+  await null;
+  assertAllDefined({
+    currentChainReference,
+    supportsCctp,
+    log,
+    getNobleICA,
+    settlementAccount,
+    statusManager,
+    tx,
+  });
+  const { amount, destination, txHash } = tx;
+  log('trying forward for', amount, 'to', destination, 'for', txHash);
+
+  const { namespace, reference } = parseAccountId(destination);
+
+  const settlement = await settlementAccount;
+  const intermediateRecipient = getNobleICA().getAddress();
+
+  if (reference === currentChainReference) {
+    await settlement.send(destination, amount);
+    statusManager.forwarded(txHash);
+  } else if (namespace === 'cosmos') {
+    try {
+      await settlement.transfer(destination, amount, {
+        timeoutRelativeSeconds: FORWARD_TIMEOUT.sec,
+        forwardOpts: {
+          intermediateRecipient,
+          timeout: FORWARD_TIMEOUT.p,
+        },
+      });
+      log('forward retry successful for', txHash);
+      statusManager.forwarded(tx.txHash);
+    } catch (reason) {
+      // funds remain in `settlementAccount` and must be recovered via a
+      // contract upgrade
+      log('🚨 forward transfer rejected!', reason, txHash);
+      // update status manager, flagging a terminal state that needs to be
+      // manual intervention or a code update to remediate
+      statusManager.forwardFailed(txHash);
+    }
+  } else if (supportsCctp(destination)) {
+    try {
+      await settlement.transfer(intermediateRecipient, amount, {
+        timeoutRelativeSeconds: FORWARD_TIMEOUT.sec,
+      });
+    } catch (reason) {
+      // funds remain in `settlementAccount` and must be recovered via a
+      // contract upgrade
+      log('🚨 forward intermediate transfer rejected!', reason, txHash);
+      // update status manager, flagging a terminal state that needs manual
+      // intervention or a code update to remediate
+      statusManager.forwardFailed(txHash);
+    }
+
+    try {
+      await getNobleICA().depositForBurn(destination, amount);
+      log('forward transfer and depositForBurn successful for', txHash);
+      statusManager.forwarded(tx.txHash);
+    } catch (reason) {
+      // funds remain in `nobleAccount` and must be recovered via a
+      // contract upgrade
+      log('🚨 forward depositForBurn rejected!', reason, txHash);
+      // update status manager, flagging a terminal state that needs manual
+      // intervention or a code update to remediate
+      statusManager.forwardFailed(txHash);
```

Comment:
I know this is just duplication of the code in `depositForBurnHandler.onRejected`, but this seems like a different failed state that the others where the funds are no longer in the local chain. I assume this will require a different retry mechanism?

Reply:
yes. retry it out of scope for this PR